src/xds/client.rs

```diff
@@ -266,6 +265,26 @@ impl AdsClient {
                 tokio::time::sleep(backoff).await;
                 backoff
             }
+            Err(e @ Error::GrpcStatus(_)) => {
+                let err_detail = e.to_string();
+                let status: tonic::Status = e.try_into().unwrap();
+                if status.code() == tonic::Code::Unknown
+                    || status.code() == tonic::Code::Cancelled
+                    || status.code() == tonic::Code::DeadlineExceeded
+                    || (status.code() == tonic::Code::Unavailable
+                        && status.message().contains("transport is closing"))
+                {
+                    debug!("XDS client terminated: {}, retrying", err_detail);
+                    self.metrics
+                        .increment(&ConnectionTerminationReason::Reconnect);
+                } else {
+                    warn!("XDS client error: {}, retrying", err_detail);
+                    self.metrics.increment(&ConnectionTerminationReason::Error);
+                }
+                // For gRPC errors, we connect immediately
```

Comment:
Its not clear to me why we don't want backoff here but we do for Connection. Does envoy do the same?

Reply:
This is just same as previous, i do not change it.  

@ramaraochavali Do you know envoy's behavior

Reply:
Envoy there is a back off https://github.com/envoyproxy/envoy/blob/a4053922d7311e11c18e764ff5e5bfa8f71cd334/source/common/config/grpc_stream.h#L21 and there are efforts to make it configurable https://github.com/envoyproxy/envoy/pull/24701

Reply:
Thanks， suggest open a new issue to tack that

Reply:
https://github.com/istio/ztunnel/issues/413